rspace/src/main/scala/coop/rchain/rspace/examples/AddressBookExample.scala

```diff
@@ -26,7 +25,7 @@ object AddressBookExample {
 
   /* Here we define a type for patterns */
 
-  sealed trait Pattern                       extends Product with Serializable
+  sealed trait Pattern                       extends Serializable
```

Comment:
`extends Product with Serializable` is a "trick" to get better type inference from sealed trait + case class sums.

In this case, it also happens that we **want** to extend `Serializable` as well...

Currently:
```scala
scala> val nameMatch = NameMatch("Henry")
nameMatch: coop.rchain.rspace.examples.AddressBookExample.NameMatch = NameMatch(Henry)

scala> val cityMatch = CityMatch("Lowell")
cityMatch: coop.rchain.rspace.examples.AddressBookExample.CityMatch = CityMatch(Lowell)

scala> val matches = List(nameMatch, cityMatch)
matches: List[coop.rchain.rspace.examples.AddressBookExample.Pattern] = List(NameMatch(Henry), CityMatch(Lowell))
```

Without extending `Product`:
```scala
scala> val nameMatch = NameMatch("Henry")
nameMatch: coop.rchain.rspace.examples.AddressBookExample.NameMatch = NameMatch(Henry)

scala> val cityMatch = CityMatch("Lowell")
cityMatch: coop.rchain.rspace.examples.AddressBookExample.CityMatch = CityMatch(Lowell)

scala> val matches = List(nameMatch, cityMatch)
matches: List[coop.rchain.rspace.examples.AddressBookExample.Pattern with Product] = List(NameMatch(Henry), CityMatch(Lowell))
```

There's a good write-up of this phenomenon here:
https://underscore.io/blog/posts/2015/06/04/more-on-sealed.html